shopify_function_macro/src/lib.rs

```diff
@@ -114,6 +115,169 @@ pub fn shopify_function(
+    export: Option<LitStr>,
+    query_path: Option<LitStr>,
+    schema_path: Option<LitStr>,
+    input_stream: Option<Expr>,
+    output_stream: Option<Expr>,
+}
+
+impl ShopifyFunctionTargetArgs {
+    fn parse<K: syn::parse::Parse, V: syn::parse::Parse>(
+        input: &ParseStream<'_>,
+    ) -> syn::Result<V> {
+        input.parse::<K>()?;
+        input.parse::<Token![=]>()?;
+        let value: V = input.parse()?;
+        if input.lookahead1().peek(Token![,]) {
+            input.parse::<Token![,]>()?;
+        }
+        Ok(value)
+    }
+}
+
+impl Parse for ShopifyFunctionTargetArgs {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let mut args = Self::default();
+        while !input.is_empty() {
+            let lookahead = input.lookahead1();
+            if lookahead.peek(kw::export) {
+                args.export = match Self::parse::<kw::export, LitStr>(&input) {
+                    Ok(export) => Some(export),
+                    _ => None,
+                };
+            } else if lookahead.peek(kw::query_path) {
+                args.query_path = Some(Self::parse::<kw::query_path, LitStr>(&input)?);
+            } else if lookahead.peek(kw::schema_path) {
+                args.schema_path = Some(Self::parse::<kw::schema_path, LitStr>(&input)?);
+            } else if lookahead.peek(kw::input_stream) {
+                args.input_stream = Some(Self::parse::<kw::input_stream, Expr>(&input)?);
+            } else if lookahead.peek(kw::output_stream) {
+                args.output_stream = Some(Self::parse::<kw::output_stream, Expr>(&input)?);
+            } else {
+                panic!("Unknown shopify_function_target parameter");
+            }
+        }
+        Ok(args)
+    }
+}
+
+fn extract_shopify_function_return_type(ast: &syn::ItemFn) -> Result<&syn::Ident, syn::Error> {
+    use syn::*;
+
+    let ReturnType::Type(_arrow, ty) = &ast.sig.output else {
+        return Err(Error::new_spanned(&ast.sig, "Shopify Functions require an explicit return type"))
+    };
+    let Type::Path(path) = ty.as_ref() else {
+        return Err(Error::new_spanned(&ast.sig, "Shopify Functions must return a Result"))
+    };
+    let result = path.path.segments.last().unwrap();
+    if result.ident != "Result" {
+        return Err(Error::new_spanned(
+            result,
+            "Shopify Functions must return a Result",
+        ));
+    }
+    let PathArguments::AngleBracketed(generics) = &result.arguments else {
+        return Err(Error::new_spanned(result, "Shopify Function Result is missing generic arguments"))
+    };
+    if generics.args.len() != 1 {
+        return Err(Error::new_spanned(
+            generics,
+            "Shopify Function Result takes exactly one generic argument",
+        ));
+    }
+    let GenericArgument::Type(ty) = generics.args.first().unwrap() else {
+        return Err(Error::new_spanned(generics, "Shopify Function Result expects a type"))
```

Comment:
Not sure how to get this one. I got a different error when I tried a type that doesn't exist:

```
error: proc-macro derive panicked
 --> example_with_targets/src/lib.rs:5:1
  |
5 | #[shopify_function_target(query_path = "a.graphql", schema_path = "schema.graphql")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: graphql-client-codegen internal error: failed to resolve TypeId for `Something°.
```